composer/algorithms/gated_linear_units/gated_linear_units.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+import warnings
+from typing import Callable, Dict, Optional, Sequence, Type, Union
+
+import torch
+
+try:
+    from transformers.models.bert.modeling_bert import BertIntermediate, BertOutput
+    TRANSFORMERS_INSTALLED = True
+except ImportError as e:
+    TRANSFORMERS_INSTALLED = False
+
+from composer.algorithms.gated_linear_units.gated_linear_unit_layers import BERTGatedOutput, DummyBERTIntermediateOutput
+from composer.algorithms.warnings import NoEffectWarning
+from composer.core import Algorithm, Event, State
+from composer.loggers import Logger
+from composer.models import BERTModel
+from composer.utils import check_if_transformers_installed, module_surgery
+
+log = logging.getLogger(__name__)
+
+
+def from_BertOutput(layer: torch.nn.Module,
+                    module_index: int,
+                    act_fn: Callable,
+                    wi_0_bias: bool = False,
+                    wi_1_bias: bool = False) -> BERTGatedOutput:
+    """Defines a replacement policy from a `transformers.models.bert.modeling_bert.BertOutput` to a `composer.algorithms.gated_linear_units.gated_linear_unit_layers.BERTGatedOutput`"""
+    assert isinstance(
```

Comment:
This assert isn't needed once with the change above

Reply:
Wait, for my education, does typing perform runtime checks? I thought it was just static analysis?

Reply:
Just static analysis. But this is already guaranteed by the replacement policy below, right? If someone wants to use this layer in some odd context, or where they wrote their own layer that should be replaced, I think we should allow it.